lib/utils.js

```diff
@@ -16,6 +16,46 @@ const execa = require('execa')
 const archiver = require('archiver')
 const debug = require('debug')('aio-app-scripts:utils')
 
+const ioruntime = require('@adobe/aio-cli-plugin-runtime')
+
+async function deployWsk (packageName, manifestPath, manifestContent, ow, logger) {
+  const entities = ioruntime.processPackage(manifestContent.packages, {}, {}, {})
+  await ioruntime.syncProject(packageName, manifestPath, manifestContent, entities, ow, logger)
+}
+
+async function undeployWsk (packageName, manifestContent, ow, logger) {
+  // 1. make sure that the package exists
+  let deployedPackage
+  try {
+    deployedPackage = await ow.packages.get(packageName)
+  } catch (e) {
+    if (e.statusCode === 404) throw new Error(`cannot undeploy actions for package ${packageName}, as it was not deployed.`)
+    throw e
+  }
+
+  // 2. extract deployment entities from previous deployment, this extracts all ow resources that are annotated with the
+  //    package name
+  const entities = await ioruntime.getProjectEntities(packageName, false, ow)
+
+  // 3. make sure that we also clean all actions in the main package that are not part of a cna deployment (e.g. wskdebug actions)
+  //    the goal is to prevent 409s on package delete (non empty package)
+  // note that entities.actions may contain actions outside deployedPackage
+  const actionNames = new Set(entities.actions.map(a => a.name))
+  deployedPackage.actions.forEach(a => {
+    const deployedActionName = `${packageName}/${a.name}`
+    if (!actionNames.has(deployedActionName)) {
+      entities.actions.push({ name: deployedActionName })
+    }
+  })
+
+  // 4. add apis to undeployment, apis are not part of the managed whisk project as they don't support annotations and
+  //    hence can't be retrieved with getProjectEntities + api delete is idempotent so no risk of 404s
+  entities.apis = ioruntime.processPackage(manifestContent.packages, {}, {}, {}, true).apis
```

Comment:
I know that processPackage is not async, but there is a disparity with deployPackage, and undeployPackage (similar in name) which are async -- perhaps ioruntime should make processPackage async as well to be consistent  (might be an API breakage, so not sure it's worth it)

Reply:
Hmm I see your point, but processPackage does no I/O just object manipulations.. + with correct autocompletion devs should be able to identify that the returned value is not a promise.

I guess we could do those changes once we expose a proper runtime SDK, this is just a tmp solution anyways

Reply:
that is true I didn't inspect if further to check if it would block -- thus async won't help here anyway. My only comment here is coherence of the API, since we mix and match. `processPackage` is a bear of a function anyway, with 210 lines. It is potentially a maintenance challenge already :)

Reply:
yeah this is a good point and I agree, just saying that for now the runtime api is not a real api and that there is a lot more to be done anyways, I didn't wanted to refactor the whole runtime-helper now 😉